src/main/java/org/altbeacon/beacon/service/BeaconService.java

```diff
@@ -329,6 +339,7 @@ private PendingIntent getRestartIntent() {
     /**
      * methods for clients
      */
+    @MainThread
```

Comment:
The scanner classes are not thread safe. They are started / run from the beacon service which runs from the main thread. Additionally, in the scanner class `mHandler` is defined on the main looper. One of the deferred tasks which are queued on this handler call `scanLeDevice`. As things are right now, for all intents and purposes these methods need to run on the main thread.

While we could add this to the javadoc, the annotation documents this with the benefit of giving an extra hint to intellisense. Now if you write the following you'll get a lint warning/error depending on the IDE configuration:

```java
mScanHandler.post(new Runnable() {
    @WorkerThread
    @Override
    public void run() {
        scanLeDevice(true);
    }
}
```